TabNavigator.js

```diff
@@ -95,21 +95,31 @@ export default class TabNavigator extends React.Component {
   }
 
   _renderTab(item) {
-    let icon;
     if (item === null) {
       return;
     }
-    if (item.props.selected) {
+
+    let icon;
+    let selectedIcon;
+    if (item.props.renderIcon) {
+      // Always render both icons, using opacity to control which one is visible
+      // this prevents flickering when using images in Android
+
+      icon = item.props.renderIcon();
       if (item.props.renderSelectedIcon) {
-        icon = item.props.renderSelectedIcon();
-      } else if (item.props.renderIcon) {
-        let defaultIcon = item.props.renderIcon();
-        icon = React.cloneElement(defaultIcon, {
-          style: [defaultIcon.props.style, styles.defaultSelectedIcon],
+        selectedIcon = item.props.renderSelectedIcon();
+      } else {
+        selectedIcon = React.cloneElement(icon, {
+          style: [icon.props.style, styles.defaultSelectedIcon],
         });
       }
-    } else if (item.props.renderIcon) {
-      icon = item.props.renderIcon();
+
+      selectedIcon = React.cloneElement(selectedIcon, {
+        style: [
+          styles.absolute,
```

Comment:
This doesn't really feel like a backwards compatible since people might rely on this not being absolute drawn. 
Shouldn't this only be applied in that case when its actually hidden? 

Plus absolute is not set to icon only the selectedIcon, can we remove it and only apply when hidden? Or it should be on both so they can be styled with consistency. Just thinking this might actually cause some problems when it comes to the badge text as well. I will ad a example in the repo to make it easier to verify this change.